include/clad/Differentiator/Compatibility.h

```diff
@@ -21,6 +21,97 @@ namespace clad_compat {
+#else
+   #define CLAD_COMPAT_CLANG17_ActOnLambdaExpr_getCurrentScope_ExtraParam(V) ,(V).getCurrentScope()
+#endif 
+
+//Clang 18 ArrayType::Normal -> ArraySizeModifier::Normal
+
+#if LLVM_VERSION_MAJOR < 18
+#define ArrayType_Normal ArrayType::Normal
+#else
+#define ArrayType_Normal ArraySizeModifier::Normal
+#endif
+
+// Compatibility helper function for creation UnresolvedLookupExpr.
+// Clang-18 extra argument knowndependent.
+//FIXME: Knowndependent set to false temporarily until known value found for initialisation.
+
+static inline Stmt* UnresolvedLookupExpr_Create(const ASTContext &Ctx, 
+                                                CXXRecordDecl *NamingClass, 
+                                                NestedNameSpecifierLoc QualifierLoc, 
+                                                SourceLocation TemplateKWLoc, 
+                                                const DeclarationNameInfo &NameInfo, 
+                                                bool RequiresADL, 
+                                                const TemplateArgumentListInfo *Args, 
+                                                UnresolvedSetIterator Begin, 
+                                                UnresolvedSetIterator End
+                                              )
+{
+
+  #if CLANG_VERSION_MAJOR < 18
+   return UnresolvedLookupExpr::Create(Ctx,
+                                       NamingClass,
+                                       QualifierLoc,
+                                       TemplateKWLoc,
+                                       NameInfo,
+                                       RequiresADL,
+                                       // They get copied again by
+                                       // OverloadExpr, so we are safe.
+                                       Args,
+                                       Begin,
+                                       End
+                                       );
+     
+  #else
+   bool KnownDependent=false;
+   return UnresolvedLookupExpr::Create(Ctx,
+                                       NamingClass,
+                                       QualifierLoc,
+                                       TemplateKWLoc,
+                                       NameInfo,
+                                       RequiresADL,
+                                       // They get copied again by
+                                       // OverloadExpr, so we are safe.
+                                       Args,
+                                       Begin,
+                                       End,
+                                       KnownDependent
+                                       );
+  #endif
+
+}
+
+//Clang 18 ETK_None -> ElaboratedTypeKeyword::None
+
+#if LLVM_VERSION_MAJOR < 18
+#define ElaboratedTypeKeyword_None ETK_None
+#else
+#define ElaboratedTypeKeyword_None ElaboratedTypeKeyword::None
+#endif
```

Comment:
```suggestion
#if LLVM_VERSION_MAJOR < 18
const auto ElaboratedTypeKeyword_None ETK_None
#else
const auto ElaboratedTypeKeyword_None ElaboratedTypeKeyword::None
#endif
```
We use where necessary as `clad_compat::ElaboratedTypeKeyword_None`.